app/src/main/java/dcom/taneja/dhruv/floater/screenshotservice/ScreenshotActivity.java

```diff
@@ -0,0 +1,134 @@
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.screenshot_activity);
+        Button updateButton = (Button) findViewById(R.id.screenshotButton);
+        updateButton.setOnClickListener(new View.OnClickListener() {
+
+            @Override
+            public void onClick(View view) {
+                updateTimer();
+                takeScreenshot();
+            }
+        });
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode,
+                                           String permissions[], int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+            Log.v(TAG, "Permission: " + permissions[0] + "was " + grantResults[0]);
+            //resume tasks needing this permission
+        }
+
+    }
+    private void updateTimer() {
+        TextView timer = (TextView) findViewById(R.id.editText);
+        timer.setText(android.text.format.DateFormat.format("yyyy-MM-dd_hh:mm:ss", new Date()));
+    }
+
+    private void takeScreenshot() {
+        String now = new SimpleDateFormat("yyyyMMdd_HHmmss").format(Calendar.getInstance().getTime());
+
+
+        //Check and ask for permission to write and read storage here:
+        if (Build.VERSION.SDK_INT >= 23) {
+            if (checkSelfPermission(android.Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                    == PackageManager.PERMISSION_GRANTED) {
+                Log.v(TAG, "Permission is granted");
+            } else {
+                Log.v(TAG, "Permission is revoked");
+                ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, 1);
+            }
+        } else { //permission is automatically granted on sdk<23 upon installation
+            Log.v(TAG, "Permission is granted");
+        }
+
+
+        //Activity code starting here
+        try {
+            // image naming and path  to include sd card  appending name you choose for file
+            String mPath = getFilesDir().getAbsolutePath() + "/" + now + ".jpg";
+            pathToFile = mPath;
+
+            // create bitmap screen capture
+            View v1 = getWindow().getDecorView().getRootView();
+            v1.setDrawingCacheEnabled(true);
+            Bitmap bitmap = Bitmap.createBitmap(v1.getDrawingCache());
+            v1.setDrawingCacheEnabled(false);
+
+            File imageFile = new File(mPath);
+
+            FileOutputStream outputStream = new FileOutputStream(imageFile);
+            int quality = 100;
+            bitmap.compress(Bitmap.CompressFormat.JPEG, quality, outputStream);
+            outputStream.flush();
+            outputStream.close();
+            Log.i("Path to file", pathToFile);
+
+            setImage();
+        } catch (Throwable e) {
```

Comment:
be specific
what kind of exceptions can this code run into?
all of them should be handled in their own ways

there should at least be some communication to the user that screenshot failed